src/js/angular/core/services/jwt-auth.service.js

```diff
@@ -3,294 +3,302 @@ import 'angular/core/services/openid-auth.service.js';
-             * by sending a request to the dedicated API endpoint using the current authentication
-             * header. If GraphDB cannot recognize the authentication, this method will fallback
-             * to free access (if enabled) or redirect to the login page. It will emit 'securityInit'.
-             *
-             * @param {boolean} noFreeAccessFallback If true does not fallback to free access.
-             * @param {boolean} justLoggedIn Indicates that the user just logged in.
+             * Returns whether the current user has external authentication.
+             * @return {boolean} True if the user is authenticated via an external system.
              */
-            this.getAuthenticatedUserFromBackend = function(noFreeAccessFallback, justLoggedIn) {
-                SecurityRestService.getAuthenticatedUser().
-                success(function(data, status, headers) {
-                    const token = AuthTokenService.getAuthToken();
-                    if (token && token.startsWith('GDB')) {
-                        // There is a previous authentication via JWT, it's still valid
-                        // so refresh the principal
-                        that.externalAuthUser = false;
-                        that.principal = data;
-                        $rootScope.$broadcast('securityInit', that.securityEnabled, true, that.freeAccess);
-                        // console.log('previous JWT authentication ok');
-                    } else if (that.openIDEnabled && token && token.startsWith('Bearer')) {
-                        // The auth was obtained from OpenID, we need to authenticate with the returned user
-                        that.authenticate(data, token);
-                    } else {
-                        // There is no previous authentication but we got a principal via
-                        // an external authentication mechanism (e.g. Kerberos)
-                        that.externalAuthUser = true;
-                        that.authenticate(data, ''); // this will emit securityInit
-                        // console.log('external authentication ok');
-                    }
-                }).error(function () {
-                    if (noFreeAccessFallback || !that.freeAccess) {
-                        $rootScope.redirectToLogin(false, justLoggedIn);
-                    } else {
-                        that.securityInitialized = true;
-                        if (!that.hasExplicitAuthentication()) {
-                            that.clearAuthentication();
-                            // console.log('free access fallback');
-                        }
-                    }
-                });
-            };
+            const hasExternalAuthUser = () => _externalAuthUser;
 
-            this.initSecurity = function () {
-                this.securityInitialized = false;
+            /**
+             * Initializes security by retrieving the security configuration and authenticating the user.
```

Comment:
I wish it was so simple, but this function is just plain stupid and does a ton of mindbogling stuff and has a flow that is beyond my abilities to follow.